tracer/test/test-applications/debugger/Samples.Probes/SmokeTests/AsyncCallChain.cs

```diff
@@ -0,0 +1,35 @@
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+namespace Samples.Probes.SmokeTests
+{
+    internal class AsyncCallChain : IAsyncRun
```

Comment:
Great job on these tests! It looks like you covered all the trickiest edge cases, really awesome 🎉